### DISCUSSION TITLE
Welcome to tsvConverter Discussions!

### DISCUSSION THREAD

**Discussion by ajnyga** [ajnyga/tsvConverter#20] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ramanfr** [comment]
Hi @ajnyga!

We’re working on importing content for a journal whose archive is in HTML only and outside OJS. 

I realise that importing HTML galleys with dependent files would be a much bigger lift than handling PDFs, but I wanted to check whether you ever looked into importing HTML with the converter, or if you see any potential angle or workaround that might make it doable. (The option to use a remote URL for galleys is probably not ideal here, as the journal would prefer not to keep hosting the HTML on an external server.)

Any thoughts or pointers you have are appreciated!

/Ramana

**Reply by ajnyga** [reply]
Hi,

So my approach with dependent files has been that for any journal importing HTML we have a folder inside the OJS public folder. So for example public/html/journalId/. We move the image files to that folder and the refer to image url's that point to files in that folder.

This has been a working and simple solution. 

Note that we have moved to the development of this script to https://github.com/pkp/Sheet2OJS - it is a modernization of the old code and any future changes will happen there. I will be closing this repository at some point.

**Comment by ramanfr** [comment]
Thanks for this information - and for letting me know the repo has moved. 

Good to know you've successfully used the script to import HTML files! 

I just want to make sure I understand the setup correctly. Would the HTML file itself also live in the same public folder as the CSS and images, so that the whole package is in e.g. public/html/journalId/? And in that case, do you include the full URL to the HTML file directly in the Excel sheet (e.g. https://publicera.kb.se/ir/.../article.html) so that it ends up in the converted XML?

I’m also assuming the HTML files would need to be edited so that image references point to the right paths in that folder — is that how you’ve handled it?

**Reply by ajnyga** [reply]
No, the HTML files are embedded to the XML just like the PDF's and they end up in the OJS files folder like the PDF's do. So it is just the embedded files that go to the public folder.

Yes, it requires us to edit the HTML files before we import them and change the image url's to match the new path to the public folder.

**Reply by ramanfr** [reply]
Ok, thanks for the additional information!

**Reply by ajnyga** [reply]
So what you can actually do is that you transfer the dependent files in advance, edit the html and you can in practice if the linking is working by opening the files locally before importing. I do not think that even moving the article later on to some other instace of OJS is such a big deal. it is basically doing batch changes of url paths to a bunch of documents which is fairly easy when the path you are changing is always the same.

OJS 3.6 will most likely have some better features in handling the dependent files and HTML btw.
